File: package/main/src/Predicate/isNotNullish.ts
/**
 * Checks whether a value is null or undefined
 * @param {unknown} value - The value to check
 * @returns {boolean} True if the value is null or undefined
 * @example
 * isNotNullish(null); // false
 * isNotNullish(undefined); // false
 * isNotNullish(0); // true
 * isNotNullish(""); // true
 */
export const isNotNullish = <T>(
  value: T,
): value is Exclude<T, null | undefined> =>
  value !== null && value !== undefined;

File: package/umt_python/src/predicate/is_not_nullish.py
def is_not_nullish(value: object) -> bool:
    """
    Checks whether a value is not None.

    In Python, the equivalent of JavaScript's null/undefined is None.

    Args:
        value: The value to check

    Returns:
        True if the value is not None

    Example:
        >>> is_not_nullish(None)
        False
        >>> is_not_nullish(0)
        True
        >>> is_not_nullish("")
        True
    """
    return value is not None
